etc/scripts/docenizer.py

```diff
@@ -31,18 +34,31 @@
     'MOVQ',  # defined in MOVD:MOVQ
     'MOVSD'  # defined in MOVS:MOVSB:MOVSW:MOVSD:MOVSQ
 ]
+# Where to extract the asmdoc archive.
+ASMDOC_DIR = "asm-docs"
 
 
 class Instruction(object):
     def __init__(self, name, names, tooltip, body):
         self.name = name
+        self.url = "http://www.felixcloutier.com/x86/{}.html".format(urllib.quote(name))
         self.names = names
         self.tooltip = tooltip.rstrip(': ,')
         self.body = body
 
     def __str__(self):
         return "{} = {}\n{}".format(self.names, self.tooltip, self.body)
 
+def download_asm_doc_archive():
+    print("Downloading archive...")
+    urllib.urlretrieve("http://www.felixcloutier.com/x86/x86.zip", "x86.zip")
+    if os.path.isdir(ASMDOC_DIR):
```

Comment:
Can the zip package overwrite instead of removing everything? What if I download this to a folder with important documents?! :dagger:

Reply:
I had thought about this, but if we don't do it and a page gets moved, we cannot detect it. Same goes if a page is deleted. I can remove that you are afraid I delete your precious documents :imp:

Reply:
Hmmm. Maybe the check could be extended to be `.isdir(os.path.join(ASMDOC_DIR, "html")): shutil.rmtree(os.path.join(ASMDOC_DIR, "html"))` so only the actual documentation files are deleted?

Reply:
This could be extended to only remove `.html` files inside that. Files from the archive are automatically overwritten, though.

Reply:
Sounds like a good plan!

Reply:
Sounds like a good plan!

Reply:
This should be fixed in [66e1](https://github.com/mattgodbolt/compiler-explorer/pull/785/commits/66e16159c63136af90d7865a3e1a8dc3a5c55bc5).